Jetcaster/wear/src/main/java/com/example/jetcaster/ui/episode/EpisodeScreen.kt

```diff
@@ -0,0 +1,282 @@
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.wear.compose.foundation.lazy.items
+import androidx.wear.compose.material.ChipDefaults
+import androidx.wear.compose.material.MaterialTheme
+import androidx.wear.compose.material.Text
+import com.example.jetcaster.R
+import com.example.jetcaster.core.data.database.model.EpisodeToPodcast
+import com.example.jetcaster.core.data.database.model.toPlayerEpisode
+import com.example.jetcaster.core.model.PlayerEpisode
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import com.google.android.horologist.composables.PlaceholderChip
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults.listTextPadding
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults.padding
+import com.google.android.horologist.compose.layout.ScalingLazyColumnState
+import com.google.android.horologist.compose.layout.ScreenScaffold
+import com.google.android.horologist.compose.layout.rememberResponsiveColumnState
+import com.google.android.horologist.compose.material.AlertDialog
+import com.google.android.horologist.compose.material.Button
+import com.google.android.horologist.compose.material.ListHeaderDefaults
+import com.google.android.horologist.compose.material.ResponsiveListHeader
+import com.google.android.horologist.media.ui.screens.entity.EntityScreen
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+
+@Composable fun EpisodeScreen(
+    onPlayButtonClick: () -> Unit,
+    onErrorDialogCancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    episodeViewModel: EpisodeViewModel = hiltViewModel()
+) {
+    val uiState by episodeViewModel.uiState.collectAsStateWithLifecycle()
+
+    EpisodeScreen(
+        uiState = uiState,
+        onPlayButtonClick = onPlayButtonClick,
+        modifier = modifier,
+        onPlayEpisode = episodeViewModel::onPlayEpisode,
+        onAddToQueue = episodeViewModel::addToQueue,
+        onErrorDialogCancelClick = onErrorDialogCancelClick,
+    )
+}
+
+@Composable
+fun EpisodeScreen(
+    uiState: EpisodeScreenState,
+    onPlayButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    onPlayEpisode: (PlayerEpisode) -> Unit,
+    onAddToQueue: (PlayerEpisode) -> Unit,
+    onErrorDialogCancelClick: () -> Unit
+) {
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = padding(
+            first = ScalingLazyColumnDefaults.ItemType.Text,
+            last = ScalingLazyColumnDefaults.ItemType.Chip
+        )
+    )
+    ScreenScaffold(
+        scrollState = columnState,
+        modifier = modifier
+    ) {
+        when (uiState) {
+            is EpisodeScreenState.Loaded -> {
```

Comment:
Are some of these screens very similar structurally? Could the UIState represent the different screens and reuse the same screen composable?